src/Makefile

```diff
@@ -18,6 +18,9 @@ ifneq ($(LIBBPF_MAJMIN_VERSION), $(LIBBPF_MAP_VERSION))
 $(error Libbpf release ($(LIBBPF_VERSION)) and map ($(LIBBPF_MAP_VERSION)) versions are out of sync!)
 endif
 
+$(if $(findstring default,$(origin CC)),$(eval CC = $(CROSS_COMPILE)$(CC)))
+$(if $(findstring default,$(origin LD)),$(eval LD = $(CROSS_COMPILE)$(LD)))
```

Comment:
can't user just specify CC and LD like this:

```
make CC=<whatever> LD=<whatever>
```

?

Reply:
> 

Yes, you can do that when building standalone.
However, user (like me) may want to build libbpf inside other tools or applications.
F.ex., 
when building inside bpftool, it supports CROSS_COMPILE env.
User builds like:
`make CROSS_COMPILE=xxx ARCH=xxx`,  will generate an arm64 bpftool trying to link a x86 libbpf.

That's why I add this change.

Reply:
I just checked the makefile of bpftool.  It passes HOSTCC as the value of CC to build libbpf for bootstrap. But, it doesn't give a value of CC to create a target binary of libbpf.  You could fix it at the bpftool side by providing a correct CC value.

By the definitions of BUILD, HOST, and TARGET machines, as I see it, bpftool build itself for BUILD and HOST machines while HOST equals TARGET for this case.  It uses the bpftool of the BUILD machine to generate header files at build time—however, the makefile of bpftool calls bootstrap HOST, which is confusing.

Reply:
Let's follow what tools/scripts/Makefile.include is doing and let's define allow-override function and call it. Keep it similar to how kernel is handling this?